Mathlib/Algebra/CharZero/Infinite.lean

```diff
@@ -0,0 +1,24 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+
+! This file was ported from Lean 3 source module algebra.char_zero.infinite
+! leanprover-community/mathlib commit 509de852e1de55e1efa8eacfa11df0823f26f226
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.CharZero.Defs
+import Mathlib.Data.Fintype.Card
+
+/-! # A characteristic-zero semiring is infinite -/
+
+
+open Set
+
+variable (M : Type _) [AddMonoidWithOne M] [CharZero M]
+
+-- see Note [lower instance priority]
+instance (priority := 100) CharZero.Infinite : Infinite M :=
+  Infinite.of_injective Nat.cast Nat.cast_injective
```

Comment:
```suggestion
instance (priority := 100) CharZero.infinite : Infinite M :=
  Infinite.of_injective Nat.cast Nat.cast_injective
```
Note that mathport is very unlikely to make the wrong choice between the three conventions (snake_case, lowerCamelCase and UpperCamelCase). The mistakes it does make almost always come from implementing the convention wrong.